Issue#57 쿠폰 저장에 관한  동시성 제어

0han7 opened the pull request:
## 개요

1. 파트
- 스핀 락(Lock)을 활용해 쿠폰 저장 시 발생하는 동시성 제어를 구현했습니다.

## PR 유형

어떤 변경 사항이 있나요?

- [x]  새로운 기능 추가
- [x]  버그 수정
- [ ]  CSS 등 사용자 UI 디자인 변경
- [x]  코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x]  코드 리팩토링
- [x]  주석 추가 및 수정
- [ ]  문서 수정
- [x]  테스트 추가, 테스트 리팩토링
- [ ]  빌드 부분 혹은 패키지 매니저 수정
- [x]  파일 혹은 폴더명 수정
- [x]  파일 혹은 폴더 삭제

## PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x]  커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x]  변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

0han7 committed:
[#57] feature : 동시성 제어 기능 추가

0han7 committed:
[#57] chore : 코드 수정

0han7 committed:
[#57] chore : 테스트 코드 주석 처리

0han7 committed:
[#57] chore : @BatchSize 임시 추가

seongjun1130 reviewed (approved):
전체적으로 동시성을 이해하시고 적용한모습에서 발전속도가 남다르신것 같습니다.
더 나아가서 왜 해당로직에 이러한 동시성제어방식을 사용했는지 까지 설명이 가능하다면 좋을것같네요!

92jy38 reviewed (commented):
동시성 제어 저는 아직도 어렵던데 스핀락 잘 사용하신 것 같습니다 👍 
고생하셨습니다.
아 그리고 PR 유형 [x ] 버그 수정 << 요기 깨져있네요 이거두 같이 확인해주시면 좋을 것 같습니다 💯

0han7 committed:
[#57] chore : 에러코드 수정

0han7 committed:
[#57] refactor : redis Write-Back 기능 구현
- 테스트 코드 작성 후 테스트 완료

0han7 committed:
[#57] chore : 테스트 코드 주석 처리

0han7 committed:
[#57] chore : 피드백 수정

0han7 committed:
[#57] chore : 피드백 수정

0han7 committed:
[#57] chore : API 수정

0han7 committed:
[#57] chore : dto 위치 이동

0han7 committed:
[#57] chore : dto 위치 이동으로 인한 import문 추가

# Conflicts:
#	src/main/java/com/sparta/travelconquestbe/api/admin/controller/AdminController.java
#	src/main/java/com/sparta/travelconquestbe/api/admin/service/AdminService.java

0han7 committed:
[#57] refactor : redis 캐싱을 활용한 성능 개선

0han7 committed:
[#57] chore : 쿠폰 DB 동기화 로직 추가

0han7 committed:
[#57] refactor : 쿠폰 redis 로직 수정 및 테스트 코스 추가

0han7 committed:
[#57] chore : 쿠폰 reids 동기화 테스트 코드 추가

0han7 committed:
[#57] refactor : 쿠폰 스케줄러 redis 동기화 코드 수정 및 테스트 코드 추가

0han7 committed:
[#57] refactor : 쿠폰 스케줄러 redis 동기화 코드 수정 및 테스트 코드 추가

0han7 committed:
[#57] chore : AdminController, AdminService 쿠폰 dto import문 수정

0han7 committed:
[#57] chore : MyCouponServiceTest 코드 수정

0han7 committed:
[#57] chore : 테스트 코드 주석처리
